zhusuan/distributions/univariate.py

```diff
@@ -1127,3 +1121,112 @@ def _log_prob(self, given):
 
     def _prob(self, given):
         return tf.exp(self._log_prob(given))
+
+
+class BinConcrete(Distribution):
+    """
+    The class of univariate BinConcrete distribution from (Maddison, 2016).
+    It is the binary case of
+    :class:`~zhusuan.distributions.multivariate.Concrete`.
+    See :class:`~zhusuan.distributions.base.Distribution` for details.
+
+    :param temperature: A 0-D `float` Tensor. The temperature of the relaxed
+        distribution. The temperature should be positive.
+    :param logits: A `float` Tensor. The log-odds of probabilities of being 1.
+
+        .. math:: \\mathrm{logits} = \\log \\frac{p}{1 - p}
+
+    :param group_event_ndims: A 0-D `int32` Tensor representing the number of
+        dimensions in `batch_shape` (counted from the end) that are grouped
+        into a single event, so that their probabilities are calculated
+        together. Default is 0, which means a single value is an event.
+        See :class:`~zhusuan.distributions.base.Distribution` for more detailed
+        explanation.
+    :param is_reparameterized: A Bool. If True, gradients on samples from this
+        distribution are allowed to propagate into inputs, using the
+        reparametrization trick from (Kingma, 2013).
+    :param check_numerics: Bool. Whether to check numeric issues.
+    """
+
+    def __init__(self,
+                 temperature,
+                 logits,
+                 group_event_ndims=0,
+                 is_reparameterized=True,
+                 check_numerics=False):
+        self._logits = tf.convert_to_tensor(logits)
+        param_dtype = assert_same_float_dtype(
+            [(self._logits, 'BinConcrete.logits')])
+
+        self._temperature = assert_positive_integer(
+            temperature, param_dtype, 'BinConcrete.temperature')
+        if isinstance(self._temperature, (int, float)):
```

Comment:
so here it should be float right? is there actually a need to check if positive? I guess we  can just replace the check with a scalar (rank) check.